zigzag-circuit.md

```diff
@@ -0,0 +1,179 @@
+- `LAYER_CHALLENGES : [LAYERS]UInt`: Number of challenges per layer.
+- `EXPANSION_DEGREE: UInt`: Degree of each bipartite expander graph to extend dependencies between layers.
+- `BASE_DEGREE: UInt`: Degree of each Depth Robust Graph.
+- `TREE_DEPTH: UInt`: Depth of the Merkle tree. Note, this is (log_2(Size of original data in bytes)).
+- `PARENT_NODES : UInt`: Defined as `EXPANSION_DEGREE+BASE_DEGREE`.
+
+#### Public Inputs
+
+*Inputs that the prover uses to generate a SNARK proof and that the verifier uses to verify it*
+
+- `ReplicaId : Fr`: A unique identifier for the replica.
+
+- `CommD : Fr`: the Merkle tree root hash of the original data (input to the first layer).
+
+- `CommRLast : Fr`: The Merkle tree root hash of the final replica (output of the last layer).
+
+- `CommRStar : Fr`: A commitment to each `l` layer's Merkle tree root hash `CommR[l]` and `ReplicaId`.
+
+- `InclusionPath : [LAYERS][]Fr`: Inclusion path for the challenged data and replica node.
+
+  Note: Each layer `l` has `LAYER_CHALLENGES[l]` inclusion paths.
+
+- `ParentInclusionPath : [LAYERS][][PARENT_NODES]Fr`:  Inclusion path for the parent nodes of the corresponding `InclusionPath[l][c]` nodes.
+
+  Note: Each layer `l` has `LAYER_CHALLENGES[l]` inclusion paths.
+
+##### Design notes
+
+- `CommRLast` is a public input, since we will be using it during Proof-of-Spacetime
+
+#### Private Inputs
+
+*Inputs that the prover uses to generate a SNARK proof, these are not needed by the verifier to verify the proof*
+
+- `CommR : [LAYERS-1]Fr`: Commitment of the the encoded data at each layer. 
+
+  Note: Size is `LAYERS-1` since the commitment to the last layer is `CommRLast`
+
+- `DataProof : [LAYERS][][TREE_DEPTH-2]Fr`: Merkle tree inclusion proof for the current layer unencoded challenged nodes.
+
+  Note: Size of proof per layer is `TREE_DEPTH-2` because it excludes (1) the roothash, and (2) the leaf value: (1) the root hash of the proof will be the corresponding unencoded data commitment for the current layer `l` (either `CommD` or `CommR[l-1]`), (2) the leaf value is `DataValue[l][c]`
+
+- `ReplicaProof : [LAYERS][][TREE_DEPTH-2]Fr`: Merkle tree inclusion proof for the current layer encoded challenged nodes.
+
+  Note: Size of proof per layer is `TREE_DEPTH-2` because it excludes (1) the roothash, and (2) the leaf value: (1) the root hash of the proof will be the corresponding unencoded data commitment for the current layer  (either `CommR[l]` or `CommRLast`), (2) the leaf value is `ReplicaValue[l][c]`
+
+- `ParentProof : [LAYERS][][PARENT_NODES][TREE_DEPTH-2]Fr`: Pedersen hashes of the Merkle inclusion proofs of the parent nodes for each challenged node at layer `l`.
+
+  Note: Size of proof per layer is `TREE_DEPTH-2` because it excludes (1) the roothash, and (2) the leaf value: (1) the root hash of the proof will be the corresponding unencoded data commitment for the current layer  (either `CommR[l]` or `CommRLast`), (2) the leaf value is `ParentValue[l][c][p]`
```

Comment:
yeah, let's definitely collapse this 3 very similar notes if possible (I get that a spec has to be exhaustive and I don't mean to lose information in the collapse but having the first two lines of the 3 notes being identical it's very easy to pass them by not realizing the difference that lies at the end)

Reply:
fixing